Trinap/Sources/Presenter/Common/Reusable/TrinapMutiSelectionCalendarView.swift

```diff
@@ -0,0 +1,186 @@
+            return self.configureDayItemProvider(day: day, calendar: calendar)
+        }
+        .monthHeaderItemProvider { month in
+            return self.configureMonthHeaderItemProvider(month)
+        }
+    }
+    
+    func configureDayItemProvider(day: Day, calendar: Calendar) -> CalendarItemModel<DayView> {
+        var invariantViewProperties = DayView.InvariantViewProperties.baseInteractive
+        
+        if day.day < calendar.component(.day, from: Date()) {
+            invariantViewProperties = DayView.InvariantViewProperties.baseNonInteractive
+            invariantViewProperties.font = TrinapFontFamily.Pretendard.bold.font(size: 16)
+            invariantViewProperties.textColor = TrinapAsset.disabled.color
+        } else {
+            if self.selectedDate.contains(where: {
+                return calendar.component(.day, from: $0) == day.day
+            }) {
+                invariantViewProperties.font = TrinapFontFamily.Pretendard.bold.font(size: 16)
+                invariantViewProperties.textColor = TrinapAsset.white.color
+                invariantViewProperties.backgroundShapeDrawingConfig.fillColor = TrinapAsset.primary.color
+            }
+        }
+        
+        return DayView.calendarItemModel(
+            invariantViewProperties: invariantViewProperties,
+            viewModel: .init(
+                dayText: "\(day.day)",
+                accessibilityLabel: nil,
+                accessibilityHint: nil
+            )
+        )
+    }
+    
+    func configureMonthHeaderItemProvider(_ month: Month) -> CalendarItemModel<MonthHeaderView> {
+        var invariantViewProperties = MonthHeaderView.InvariantViewProperties.base
+        let trinapDoubleOffset = self.trinapOffset * 2
+        invariantViewProperties.edgeInsets = .init(
+            top: trinapDoubleOffset,
+            leading: trinapDoubleOffset,
+            bottom: 0,
+            trailing: 0
+        )
+        invariantViewProperties.font = TrinapFontFamily.Pretendard.bold.font(size: 20)
+        return MonthHeaderView.calendarItemModel(
+            invariantViewProperties: invariantViewProperties,
+            viewModel: .init(
+                monthText: "\(month.year)년 \(month.month)월",
+                accessibilityLabel: nil
+            )
+        )
+    }
+    
+    func filterPreviousDate(with possibleDate: [Date]) -> [Date] {
+        return possibleDate.filter { date in
+            let year = self.calendar.component(.year, from: date)
+            
+            return self.calendar.component(.year, from: Date()) == year
+        }
+        .filter { date in
+            let month = self.calendar.component(.month, from: date)
+            
+            return self.calendar.component(.month, from: Date()) == month
+        }
+        .filter { date in
+            let day = self.calendar.component(.day, from: date)
+            
+            return self.calendar.component(.day, from: Date()) <= day
+        }
```

Comment:
filter을 하나로 합치는건 어떨까요?

하나의 filter마다 O(n)이 소요될텐데, 한 번 돌 때 모두 처리하는 것이 조금 더 효율적일 것 같습니다!

```suggestion
        return possibleDate.filter { date in
            let year = self.calendar.component(.year, from: date)
            let month = self.calendar.component(.month, from: date)
            let day = self.calendar.component(.day, from: date)

            return (
                self.calendar.component(.year, from: Date()) == year &&
                self.calendar.component(.month, from: Date()) == month &&
                self.calendar.component(.day, from: Date()) <= day
            )
        }
```

Reply:
혹은 `dateComponents([.year, .month, .day], from: Date())`를 사용하는 것도 좋을 것 같습니다!

Reply:
수정.